spec/controllers/viewer_controller_spec.rb

```diff
@@ -51,11 +51,75 @@
+          user = FactoryBot.create(:user)
+          allow(CDL::EligibleItemService).to receive(:item_ids).with(source_metadata_identifier: "123456").and_return(["12345"])
+          stub_bibdata(bib_id: "123456")
+          resource = FactoryBot.create_for_repository(:complete_private_scanned_resource, source_metadata_identifier: "123456")
+          sign_in user
+
+          get :auth, params: { id: resource.id.to_s }
+
+          expect(response).to be_successful
+          expect(response.body).to have_content "This Item may be protected by third-party copyright and/or related intellectual property rights."
+          expect(response.body).to have_button "Check Out"
+        end
+      end
+      context "when the item is checked out to the user" do
+        it "redirects to the viewer" do
+          user = FactoryBot.create(:user)
+          stub_bibdata(bib_id: "123456")
+          resource = FactoryBot.create_for_repository(:complete_private_scanned_resource, source_metadata_identifier: "123456")
+          allow(CDL::EligibleItemService).to receive(:item_ids).and_return(["1"])
+          charged_items = [
+            CDL::ChargedItem.new(item_id: "1", netid: user.uid, expiration_time: Time.current + 3.hours)
+          ]
+          FactoryBot.create_for_repository(:resource_charge_list, resource_id: resource.id, charged_items: charged_items)
+          sign_in user
+
+          get :auth, params: { id: resource.id.to_s }
+
+          expect(response).to redirect_to viewer_index_path(anchor: "?manifest=http://www.example.com/concern/scanned_resources/#{resource.id}/manifest")
+        end
+      end
+      context "when the item is unavailable" do
+        it "displays a copyright statement, a disabled check-out button, and an ETA in hours" do
+          user = FactoryBot.create(:user)
+          stub_bibdata(bib_id: "123456")
+          resource = FactoryBot.create_for_repository(:complete_private_scanned_resource, source_metadata_identifier: "123456")
+          allow(CDL::EligibleItemService).to receive(:item_ids).and_return(["1"])
+          charged_items = [
+            CDL::ChargedItem.new(item_id: "1", netid: "other", expiration_time: Time.current + 3.hours)
+          ]
+          FactoryBot.create_for_repository(:resource_charge_list, resource_id: resource.id, charged_items: charged_items)
+          sign_in user
+
+          get :auth, params: { id: resource.id.to_s }
+
+          expect(response).to be_successful
+          expect(response.body).to have_content "This Item may be protected by third-party copyright and/or related intellectual property rights."
+          expect(response.body).to have_button "Check Out", disabled: true
+          expect(response.body).to have_content "This item is currently checked out. The estimated wait time is about 3 hours."
+        end
+      end
+    end
+    context "when the user is logged in and the resource is private and CDL ineligible" do
+      it "redirects back to the viewer" do
```

Comment:
The viewer doesn't show it though, right? If it's private?

Reply:
Right. It'll just show a blank page. There's no valid way to get to this state.

Reply:
Can we add this as a comment to the test and/or the code?